internal/controller/promotion/argocd.go

```diff
@@ -215,9 +219,46 @@ func (a *argoCDMechanism) doSingleUpdate(
 	}
 	logging.LoggerFromContext(ctx).WithField("app", app.Name).
 		Debug("patched Argo CD Application")
+
+	a.logAppEventFn(ctx, app, "Promotion", "Promotion triggered a sync of this Application resource.")
```

Comment:
Do we want this message to mimic the message Argo CD itself would emit (https://github.com/argoproj/argo-cd/blob/44894e9e438bca5adccf58d2f904adc63365805c/server/application/application.go#L1887-L1895), or do we want it to be Kargo specific?

Reply:
While I do agree that's easier for Event consumers, it does feel like that underlines the current shortcoming in Argo CD itself which @krancour took note of in https://github.com/akuity/kargo/issues/1845#issuecomment-2059138876.

Reply:
Still not an expert on k8s events yet, but there must be some way to distinguish that, even though it looks just like an Argo CD event, Kargo wrote it, right?

IF that is so the event already has a metaphorical asterisk next to it.

Reply:
You can't really, as you produce the event _for_ an involved object. Which is the Argo CD Application in this case.

The only sign we give people is the injection of an `kargo-controller` user at the beginning of the message string.

Reply:
So basically the only telltale sign that it came from us and not from Argo CD itself is in an unstructured field?

Reply:
That's correct.